src/test/ui/parser/doc-comment-in-if-statement.stderr

```diff
@@ -1,10 +1,19 @@
-error: expected `{`, found doc comment `/*!*/`
+error: expected outer doc comment
   --> $DIR/doc-comment-in-if-statement.rs:2:13
    |
 LL |     if true /*!*/ {}
-   |     --      ^^^^^ expected `{`
-   |     |
-   |     this `if` expression has a condition, but no block
+   |             ^^^^^
+   |
+   = note: inner doc comments like this (starting with `//!` or `/*!`) can only appear before items
```

Comment:
I feel the message is a bit misleading, as it should be closer to "found invalid inner doc comment"/"inner doc comments like this (...) can only appear at the `mod` level before all items", or something like that.

Reply:
Other than this nitpick, r=me

Reply:
I believe the message is actually accurate, because `trait Foo { /*! foo */ }` is legal. In other words, this applies to *item containers* in general (but there are other contexts where inner attributes are allowed as well, e.g., `Foo { /*! foo */ };`).

Anyways, this is a pre-existing wording, and I would prefer not to inflate the PR more... :)